### PR TITLE
465 discogs: Fetch a few more metadata fields

### PR DESCRIPTION
**Pull request by thedevilisinthedetails**

### CODE REVIEW

**Comment by sampsyo** [comment]
Looks good at a high level so far! Can you please leave a comment with some background when you're ready for a review?

**Comment by thedevilisinthedetails** [comment]
Sure. So I'm grabbing the additional fields (release date, genre and release id) mentioned in the issue and storing them in the db. For the release I just extracted it from the URI. Genre behaves the same as style (for consistency) and just grabbing the release date.

File: beetsplug/discogs.py [hunk 1]
```diff
@@ -341,18 +344,25 @@ def get_album_info(self, result):
...
+                         data_url=data_url,
+                         discogs_release_id=discogs_release_id,
+                         released_date=released_date)
 
-    def format_style(self, style):
-        if style is None:
-            self._log.debug('Style not Found')
+    def format(self, classification):
+        if classification is None:
+            self._log.debug('Classification not Found')
```
Comment (sampsyo) [inline beetsplug/discogs.py]: This log message is now not very useful… maybe we can just remove it?

File: beetsplug/discogs.py [hunk 2]
```diff
@@ -341,18 +344,25 @@ def get_album_info(self, result):
...
         else:
-            return self.config['separator'].as_str().join(sorted(style))
+            return self.config['separator'].as_str()\
+                .join(sorted(classification))
+
+    def extract_release_id(self, uri):
+        if uri:
+            return uri.split("/")[-1]
```
Comment (sampsyo) [inline beetsplug/discogs.py]: It's good style to explicitly `return None` if the function's intent is to sometimes return `None`.

**Comment by thedevilisinthedetails** [comment]
addressed your comments @sampsyo

**Comment by sampsyo** [comment]
Awesome; thank you!! Merged with a changelog entry. :sparkles:

**Comment by Freso** [comment]
Based on the code, it doesn’t seem like this stops writing Discogs ID(s) as MusicBrainz ones? Any chance you could fix that as well? :pray: https://github.com/beetbox/beets/issues/604

**Comment by thedevilisinthedetails** [comment]
> Based on my quick skimming of the code, it doesn’t seem like this stops writing Discogs ID(s) as MusicBrainz ones? Any chance you could fix that as well? 🙏 #604

You sure? When I import using discogs my files are tagged via discogs metadata values. I mean I see that correct values being added in my db

**Pull request merged**